overlays/azure-cloud-overlay.yaml

```diff
@@ -0,0 +1,22 @@
+description: Charmed Kubernetes overlay to add native Azure support.
+applications:
+  kubernetes-control-plane:
+    options:
+      register-with-taints: ""
```

Comment:
why reset the [default taints](https://github.com/charmed-kubernetes/charm-kubernetes-control-plane/blob/main/config.yaml#L336)?  as written, this overlay will schedule everything with k-c-p as a valid k8s node, which is not typically what we want.

Reply:
the `azure-cloud-provider` (and other *-cloud-provider) charms i've been working on all target the control plane nodes for the controllers.  This is similar to how the current addons work -- in that they run on the control plane nodes. 

I'd welcome a discussion on this matter.

Reply:
The end result of this discussion is that `register-with-taints` shouldn't be blanked, but should instead drive some relation data in `interface-kube-control` to notify `azure-cloud-provider` of the configured taints so it can set deployment/pod tolerations appropriately